parameters/default_parameters11_03_2017.txt
show








FORMAT_VERSION: 13
SYSID_SW_TYPE: 0
SYSID_THISMAV: 1
SYSID_MYGCS: 255
CLI_ENABLED: 1
SERIAL0_BAUD: 115
SERIAL1_PROTOCOL: 1
SERIAL1_BAUD: 57
SERIAL2_PROTOCOL: 1
SERIAL2_BAUD: 57
SERIAL3_PROTOCOL: 5
SERIAL3_BAUD: 38
SERIAL4_PROTOCOL: 5
SERIAL4_BAUD: 38
AUTOTUNE_LEVEL: 6
TELEM_DELAY: 0
GCS_PID_MASK: 0
KFF_RDDRMIX: 0.500000
KFF_THR2PTCH: 0.000000
STAB_PITCH_DOWN: 2.000000
GLIDE_SLOPE_MIN: 15
GLIDE_SLOPE_THR: 5.000000
STICK_MIXING: 1
AUTO_FBW_STEER: 0
TKOFF_THR_MINSPD: 0.000000
TKOFF_THR_MINACC: 0.000000
TKOFF_THR_DELAY: 2
TKOFF_TDRAG_ELEV: 0
TKOFF_TDRAG_SPD1: 0.000000
TKOFF_ROTATE_SPD: 0.000000
TKOFF_THR_SLEW: 0
FBWA_TDRAG_CHAN: 0
LEVEL_ROLL_LIMIT: 5
LAND_PITCH_CD: 0
LAND_FLARE_ALT: 3.000000
LAND_FLARE_SEC: 2.000000
LAND_DISARMDELAY: 20
LAND_ABORT_THR: 0
NAV_CONTROLLER: 1
ALT_MIX: 1.000000
ALT_CTRL_ALG: 0
ALT_OFFSET: 0
WP_RADIUS: 90
WP_MAX_RADIUS: 0
WP_LOITER_RAD: 60
FENCE_ACTION: 0
FENCE_TOTAL: 0
FENCE_CHANNEL: 0
FENCE_MINALT: 0
FENCE_MAXALT: 0
FENCE_RETALT: 0
FENCE_AUTOENABLE: 0
FENCE_RET_RALLY: 0
STALL_PREVENTION: 1
ARSPD_FBW_MIN: 9
ARSPD_FBW_MAX: 22
FBWB_ELEV_REV: 0
FBWB_CLIMB_RATE: 2
THR_MIN: 0
THR_MAX: 75
TKOFF_THR_MAX: 0
THR_SLEWRATE: 100
FLAP_SLEWRATE: 75
THR_SUPP_MAN: 0
THR_PASS_STAB: 0
THR_FAILSAFE: 1
THR_FS_VALUE: 950
TRIM_THROTTLE: 45
THROTTLE_NUDGE: 1
FS_SHORT_ACTN: 0
FS_SHORT_TIMEOUT: 1.500000
FS_LONG_ACTN: 0
FS_LONG_TIMEOUT: 5.000000
FS_BATT_VOLTAGE: 0.000000
FS_BATT_MAH: 0.000000
FS_GCS_ENABL: 0
FLTMODE_CH: 6
FLTMODE1: 11
FLTMODE2: 11
FLTMODE3: 5
FLTMODE4: 5
FLTMODE5: 0
FLTMODE6: 0
INITIAL_MODE: 0
LIM_ROLL_CD: 4500
LIM_PITCH_MAX: 2000
LIM_PITCH_MIN: -2500
ACRO_ROLL_RATE: 180
ACRO_PITCH_RATE: 180
ACRO_LOCKING: 0
GROUND_STEER_ALT: 0.000000
GROUND_STEER_DPS: 90
TRIM_AUTO: 0
SYS_NUM_RESETS: 4
LOG_BITMASK: 65535
RST_SWITCH_CH: 0
RST_MISSION_CH: 0
TRIM_ARSPD_CM: 1200
SCALING_SPEED: 15.00000
MIN_GNDSPD_CM: 0
TRIM_PITCH_CD: 0
ALT_HOLD_RTL: 10000
ALT_HOLD_FBWCM: 0
MAG_ENABLE: 1
HIL_MODE: 0
HIL_SERVOS: 0
HIL_ERR_LIMIT: 5.000000
RTL_AUTOLAND: 0
TRIM_RC_AT_START: 0
CRASH_DETECT: 0
GND_ABS_PRESS: 101248.2
GND_TEMP: 30.99800
GND_ALT_OFFSET: 0.000000
GND_PRIMARY: 0
GPS_TYPE: 1
GPS_TYPE2: 0
GPS_NAVFILTER: 8
GPS_AUTO_SWITCH: 1
GPS_MIN_DGPS: 100
GPS_SBAS_MODE: 2
GPS_MIN_ELEV: -100
GPS_INJECT_TO: 127
GPS_SBP_LOGMASK: -256
GPS_RAW_DATA: 0
GPS_GNSS_MODE: 0
GPS_SAVE_CFG: 0
ARMING_REQUIRE: 1
ARMING_CHECK: 1
ARMING_RUDDER: 1
RNGFND_LANDING: 0
ADSB_ENABLE: 0
ADSB_BEHAVIOR: 0
RLL2SRV_TCONST: 0.500000
RLL2SRV_P: 0.400000
RLL2SRV_D: 0.020000
RLL2SRV_I: 0.040000
RLL2SRV_RMAX: 0
RLL2SRV_IMAX: 3000
RLL2SRV_FF: 0.000000
PTCH2SRV_TCONST: 0.500000
PTCH2SRV_P: 0.400000
PTCH2SRV_D: 0.020000
PTCH2SRV_I: 0.040000
PTCH2SRV_RMAX_UP: 0
PTCH2SRV_RMAX_DN: 0
PTCH2SRV_RLL: 1.000000
PTCH2SRV_IMAX: 3000
PTCH2SRV_FF: 0.000000
YAW2SRV_SLIP: 0.000000
YAW2SRV_INT: 0.000000
YAW2SRV_DAMP: 0.000000
YAW2SRV_RLL: 1.000000
YAW2SRV_IMAX: 1500
STEER2SRV_TCONST: 0.750000
STEER2SRV_P: 1.800000
STEER2SRV_I: 0.200000
STEER2SRV_D: 0.005000
STEER2SRV_IMAX: 1500
STEER2SRV_MINSPD: 1.000000
STEER2SRV_FF: 0.000000
COMPASS_OFS_X: 0.000000
COMPASS_OFS_Y: 0.000000
COMPASS_OFS_Z: 0.000000
COMPASS_DEC: 0.000000
COMPASS_LEARN: 0
COMPASS_USE: 1
COMPASS_AUTODEC: 0
COMPASS_MOTCT: 0
COMPASS_MOT_X: 0.000000
COMPASS_MOT_Y: 0.000000
COMPASS_MOT_Z: 0.000000
COMPASS_ORIENT: 4
COMPASS_EXTERNAL: 1
COMPASS_OFS2_X: 0.000000
COMPASS_OFS2_Y: 0.000000
COMPASS_OFS2_Z: 0.000000
COMPASS_MOT2_X: 0.000000
COMPASS_MOT2_Y: 0.000000
COMPASS_MOT2_Z: 0.000000
COMPASS_PRIMARY: 0
COMPASS_OFS3_X: 0.000000
COMPASS_OFS3_Y: 0.000000
COMPASS_OFS3_Z: 0.000000
COMPASS_MOT3_X: 0.000000
COMPASS_MOT3_Y: 0.000000
COMPASS_MOT3_Z: 0.000000
COMPASS_DEV_ID: 9
COMPASS_DEV_ID2: 0
COMPASS_DEV_ID3: 0
COMPASS_USE2: 1
COMPASS_ORIENT2: 0
COMPASS_EXTERN2: 0
COMPASS_USE3: 1
COMPASS_ORIENT3: 0
COMPASS_EXTERN3: 0
COMPASS_DIA_X: 1.000000
COMPASS_DIA_Y: 1.000000
COMPASS_DIA_Z: 1.000000
COMPASS_ODI_X: 0.000000
COMPASS_ODI_Y: 0.000000
COMPASS_ODI_Z: 0.000000
COMPASS_DIA2_X: 0.000000
COMPASS_DIA2_Y: 0.000000
COMPASS_DIA2_Z: 0.000000
COMPASS_ODI2_X: 0.000000
COMPASS_ODI2_Y: 0.000000
COMPASS_ODI2_Z: 0.000000
COMPASS_DIA3_X: 0.000000
COMPASS_DIA3_Y: 0.000000
COMPASS_DIA3_Z: 0.000000
COMPASS_ODI3_X: 0.000000
COMPASS_ODI3_Y: 0.000000
COMPASS_ODI3_Z: 0.000000
COMPASS_CAL_FIT: 8.000000
SCHED_DEBUG: 0
RCMAP_ROLL: 1
RCMAP_PITCH: 2
RCMAP_THROTTLE: 3
RCMAP_YAW: 4
SR0_RAW_SENS: 1
SR0_EXT_STAT: 1
SR0_RC_CHAN: 1
SR0_RAW_CTRL: 1
SR0_POSITION: 1
SR0_EXTRA1: 1
SR0_EXTRA2: 1
SR0_EXTRA3: 1
SR0_PARAMS: 10
SR1_RAW_SENS: 16514
SR1_EXT_STAT: 23593
SR1_RC_CHAN: 16773
SR1_RAW_CTRL: -26214
SR1_POSITION: 16153
SR1_EXTRA1: 3300
SR1_EXTRA2: 0
SR1_EXTRA3: 256
SR1_PARAMS: 0
SR2_RAW_SENS: 0
SR2_EXT_STAT: 0
SR2_RC_CHAN: 0
SR2_RAW_CTRL: 0
SR2_POSITION: 0
SR2_EXTRA1: 0
SR2_EXTRA2: 17076
SR2_EXTRA3: 0
SR2_PARAMS: 0
SR3_RAW_SENS: 0
SR3_EXT_STAT: 0
SR3_RC_CHAN: 0
SR3_RAW_CTRL: 0
SR3_POSITION: 0
SR3_EXTRA1: 5936
SR3_EXTRA2: 4096
SR3_EXTRA3: 0
SR3_PARAMS: 0
INS_PRODUCT_ID: 0
INS_GYROFFS_X: 0.000000
INS_GYROFFS_Y: 0.000000
INS_GYROFFS_Z: 0.000000
INS_GYR2OFFS_X: 0.000000
INS_GYR2OFFS_Y: 0.000000
INS_GYR2OFFS_Z: 0.000000
INS_GYR3OFFS_X: 0.000000
INS_GYR3OFFS_Y: 0.000000
INS_GYR3OFFS_Z: 0.000000
INS_ACCSCAL_X: 0.000000
INS_ACCSCAL_Y: 0.000000
INS_ACCSCAL_Z: 0.000000
INS_ACCOFFS_X: 0.000000
INS_ACCOFFS_Y: 0.000000
INS_ACCOFFS_Z: 0.000000
INS_ACC2SCAL_X: 0.000000
INS_ACC2SCAL_Y: 0.000000
INS_ACC2SCAL_Z: 0.000000
INS_ACC2OFFS_X: 0.000000
INS_ACC2OFFS_Y: 0.000000
INS_ACC2OFFS_Z: 0.000000
INS_ACC3SCAL_X: 0.000000
INS_ACC3SCAL_Y: 0.000000
INS_ACC3SCAL_Z: 0.000000
INS_ACC3OFFS_X: 0.000000
INS_ACC3OFFS_Y: 0.000000
INS_ACC3OFFS_Z: 0.000000
INS_GYRO_FILTER: 20
INS_ACCEL_FILTER: 20
INS_USE: 1
INS_USE2: 1
INS_USE3: 0
INS_STILL_THRESH: 0.100000
INS_GYR_CAL: 1
AHRS_GPS_GAIN: 1.000000
AHRS_GPS_USE: 1
AHRS_YAW_P: 0.200000
AHRS_RP_P: 0.200000
AHRS_WIND_MAX: 0
AHRS_TRIM_X: 0.000000
AHRS_TRIM_Y: 0.000000
AHRS_TRIM_Z: 0.000000
AHRS_ORIENTATION: 0
AHRS_COMP_BETA: 0.100000
AHRS_GPS_MINSATS: 6
ARSPD_ENABLE: 1
ARSPD_USE: 0
ARSPD_OFFSET: 1.952411
*ARSPD_RATIO: 1.633000
*ARSPD_PIN: 4
*ARSPD_AUTOCAL: 0
*ARSPD_TUBE_ORDER: 0
*ARSPD_SKIP_CAL: 0
*NAVL1_PERIOD: 20.00000
*NAVL1_DAMPING: 0.750000
*NAVL1_XTRACK_I: 0.020000
*TECS_CLMB_MAX: 5.000000
*TECS_SINK_MIN: 2.000000
*TECS_TIME_CONST: 5.000000
*TECS_THR_DAMP: 0.500000
*TECS_INTEG_GAIN: 0.100000
*TECS_VERT_ACC: 7.000000
*TECS_HGT_OMEGA: 3.000000
*TECS_SPD_OMEGA: 2.000000
*TECS_RLL2THR: 10.00000
*TECS_SPDWEIGHT: 1.000000
*TECS_PTCH_DAMP: 0.000000
*TECS_SINK_MAX: 5.000000
*TECS_LAND_ARSPD: -1.000000
*TECS_LAND_THR: -1.000000
*TECS_LAND_SPDWGT: 1.000000
*
TECS_PITCH_MAX: 0
*TECS_PITCH_MIN: 0
*TECS_LAND_SINK: 0.250000
*TECS_LAND_TCONST: 2.000000
*TECS_LAND_DAMP: 0.500000
*TECS_LAND_PMAX: 10
*BATT_MONITOR: 4
*BATT_VOLT_PIN: 3
*BATT_CURR_PIN: 2
*BATT_VOLT_MULT: 4.092000
*BATT_AMP_PERVOLT: 16.67000
*BATT_AMP_OFFSET: 0.600000
*BATT_CAPACITY: 3300
*MIS_TOTAL: 0
*MIS_RESTART: 0
*RALLY_TOTAL: 0
*RALLY_LIMIT_KM: 5.000000
*RALLY_INCL_HOME: 0
*RSSI_TYPE: 0
*RSSI_ANA_PIN: 0
*RSSI_PIN_LOW: 0.000000
*RSSI_PIN_HIGH: 5.000000
*RSSI_CHANNEL: 0
*RSSI_CHAN_LOW: 1000
*RSSI_CHAN_HIGH: 2000
setup] 

parameters/SkyHookDisco_parameters_16_Apr2017.txt
Discovery
Change 
FBWA_MIN = 12 m/s
AHRS_ORIENTATION = 6  # YAW_270
COMPASS_ORIENT   = 6 # YAW_270


ArduPlane/config.h


// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
//
//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////
//
// WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
//
//  DO NOT EDIT this file to adjust your configuration.  Create your own
//  APM_Config.h and use APM_Config.h.example as a reference.
//
// WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
///
//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////
//
// Default and automatic configuration details.
//
// Notes for maintainers:
//
// - Try to keep this file organised in the same order as APM_Config.h.example
//

#include "defines.h"

///
/// DO NOT EDIT THIS INCLUDE - if you want to make a local change, make that
/// change in your local copy of APM_Config.h.
///
#ifdef USE_CMAKE_APM_CONFIG
#error "Dont use this!"
 #include "APM_Config_cmake.h" // <== Prefer cmake config if it exists
#else
 #include "APM_Config.h" // <== THIS INCLUDE, DO NOT EDIT IT. EVER.
#endif
///
/// DO NOT EDIT THIS INCLUDE - if you want to make a local change, make that
/// change in your local copy of APM_Config.h.
///

// Just so that it's completely clear...
#define ENABLED                 1
#define DISABLED                0

// this avoids a very common config error
#define ENABLE ENABLED
#define DISABLE DISABLED

//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////
// HARDWARE CONFIGURATION AND CONNECTIONS
//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////

#ifdef CONFIG_APM_HARDWARE
#error CONFIG_APM_HARDWARE option is depreated! use CONFIG_HAL_BOARD instead.
#endif

#ifndef MAV_SYSTEM_ID
 # define MAV_SYSTEM_ID          1
#endif

//////////////////////////////////////////////////////////////////////////////
// FrSky telemetry support
//

#ifndef FRSKY_TELEM_ENABLED
#define FRSKY_TELEM_ENABLED ENABLED
#endif

//////////////////////////////////////////////////////////////////////////////
// Optical flow sensor support
//
#ifndef OPTFLOW
 # define OPTFLOW DISABLED
#endif


#ifndef RANGEFINDER_ENABLED
# define RANGEFINDER_ENABLED DISABLED
#endif

//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////
// RADIO CONFIGURATION
//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////


//////////////////////////////////////////////////////////////////////////////
// Radio channel limits
//
// Note that these are not called out in APM_Config.h.reference.
//
#ifndef CH5_MIN
 # define CH5_MIN        1000
#endif
#ifndef CH5_MAX
 # define CH5_MAX        2000
#endif
#ifndef CH6_MIN
 # define CH6_MIN        1000
#endif
#ifndef CH6_MAX
 # define CH6_MAX        2000
#endif
#ifndef CH7_MIN
 # define CH7_MIN        1000
#endif
#ifndef CH7_MAX
 # define CH7_MAX        2000
#endif
#ifndef CH8_MIN
 # define CH8_MIN        1000
#endif
#ifndef CH8_MAX
 # define CH8_MAX        2000
#endif


#ifndef FLAP_1_PERCENT
 # define FLAP_1_PERCENT 0
#endif
#ifndef FLAP_1_SPEED
 # define FLAP_1_SPEED 0
#endif
#ifndef FLAP_2_PERCENT
 # define FLAP_2_PERCENT 0
#endif
#ifndef FLAP_2_SPEED
 # define FLAP_2_SPEED 0
#endif
//////////////////////////////////////////////////////////////////////////////
// FLIGHT_MODE
// FLIGHT_MODE_CHANNEL
//
#ifndef FLIGHT_MODE_CHANNEL
 # define FLIGHT_MODE_CHANNEL    8
#endif
#if (FLIGHT_MODE_CHANNEL != 5) && (FLIGHT_MODE_CHANNEL != 6) && (FLIGHT_MODE_CHANNEL != 7) && (FLIGHT_MODE_CHANNEL != 8)
 # error XXX
 # error XXX You must set FLIGHT_MODE_CHANNEL to 5, 6, 7 or 8
 # error XXX
#endif

#if !defined(FLIGHT_MODE_1)
 # define FLIGHT_MODE_1                  FlightMode::RTL
#endif
#if !defined(FLIGHT_MODE_2)
 # define FLIGHT_MODE_2                  FlightMode::RTL
#endif
#if !defined(FLIGHT_MODE_3)
 # define FLIGHT_MODE_3                  FlightMode::FLY_BY_WIRE_A
#endif
#if !defined(FLIGHT_MODE_4)
 # define FLIGHT_MODE_4                  FlightMode::FLY_BY_WIRE_A
#endif
#if !defined(FLIGHT_MODE_5)
 # define FLIGHT_MODE_5                  FlightMode::MANUAL
#endif
#if !defined(FLIGHT_MODE_6)
 # define FLIGHT_MODE_6                  FlightMode::MANUAL
#endif


//////////////////////////////////////////////////////////////////////////////
// THROTTLE_FAILSAFE
// THROTTLE_FS_VALUE
// SHORT_FAILSAFE_ACTION
// LONG_FAILSAFE_ACTION
#ifndef THROTTLE_FAILSAFE
 # define THROTTLE_FAILSAFE              ENABLED
#endif
#ifndef THROTTLE_FS_VALUE
 # define THROTTLE_FS_VALUE              950
#endif
#ifndef SHORT_FAILSAFE_ACTION
 # define SHORT_FAILSAFE_ACTION          0
#endif
#ifndef LONG_FAILSAFE_ACTION
 # define LONG_FAILSAFE_ACTION           0
#endif

//////////////////////////////////////////////////////////////////////////////
// AUTO_TRIM
//
#ifndef AUTO_TRIM
 # define AUTO_TRIM                              DISABLED
#endif

//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////
// STARTUP BEHAVIOUR
//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////


//////////////////////////////////////////////////////////////////////////////
// GROUND_START_DELAY
//
#ifndef GROUND_START_DELAY
 # define GROUND_START_DELAY             0
#endif

//////////////////////////////////////////////////////////////////////////////
// ENABLE ELEVON_MIXING
//
#ifndef ELEVON_MIXING
 # define ELEVON_MIXING          DISABLED
#endif
#ifndef ELEVON_REVERSE
 # define ELEVON_REVERSE     DISABLED
#endif
#ifndef ELEVON_CH1_REVERSE
 # define ELEVON_CH1_REVERSE     DISABLED
#endif
#ifndef ELEVON_CH2_REVERSE
 # define ELEVON_CH2_REVERSE     DISABLED
#endif

//////////////////////////////////////////////////////////////////////////////
// CAMERA TRIGGER AND CONTROL
//
// uses 1182 bytes of memory
#ifndef CAMERA
 # define CAMERA         ENABLED
#endif

//////////////////////////////////////////////////////////////////////////////
// MOUNT (ANTENNA OR CAMERA)
//
// uses 7726 bytes of memory on 2560 chips (all options are enabled)
#ifndef MOUNT
#define MOUNT          ENABLED
#endif

//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////
// FLIGHT AND NAVIGATION CONTROL
//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////

//////////////////////////////////////////////////////////////////////////////
// Altitude measurement and control.
//
#ifndef ALTITUDE_MIX
 # define ALTITUDE_MIX                   1
#endif


//////////////////////////////////////////////////////////////////////////////
// AIRSPEED_CRUISE
//
#ifndef AIRSPEED_CRUISE
 # define AIRSPEED_CRUISE                12 // 12 m/s
#endif
#define AIRSPEED_CRUISE_CM AIRSPEED_CRUISE*100


//////////////////////////////////////////////////////////////////////////////
// MIN_GNDSPEED
//
#ifndef MIN_GNDSPEED
 # define MIN_GNDSPEED                   0 // m/s (0 disables)
#endif
#define MIN_GNDSPEED_CM MIN_GNDSPEED*100


//////////////////////////////////////////////////////////////////////////////
// FLY_BY_WIRE_B airspeed control
//
#ifndef AIRSPEED_FBW_MIN
 # define AIRSPEED_FBW_MIN               9
#endif
#ifndef AIRSPEED_FBW_MAX
 # define AIRSPEED_FBW_MAX               22
#endif

#ifndef ALT_HOLD_FBW
 # define ALT_HOLD_FBW 0
#endif
#define ALT_HOLD_FBW_CM ALT_HOLD_FBW*100


//////////////////////////////////////////////////////////////////////////////
// Servo Mapping
//
#ifndef THROTTLE_MIN
 # define THROTTLE_MIN                   0 // percent
#endif
#ifndef THROTTLE_CRUISE
 # define THROTTLE_CRUISE                45
#endif
#ifndef THROTTLE_MAX
 # define THROTTLE_MAX                   75
#endif

//////////////////////////////////////////////////////////////////////////////
// Autopilot control limits
//
#ifndef HEAD_MAX
 # define HEAD_MAX                               45
#endif
#ifndef PITCH_MAX
 # define PITCH_MAX                              20
#endif
#ifndef PITCH_MIN
 # define PITCH_MIN                              -25
#endif
#define HEAD_MAX_CENTIDEGREE HEAD_MAX * 100
#define PITCH_MAX_CENTIDEGREE PITCH_MAX * 100
#define PITCH_MIN_CENTIDEGREE PITCH_MIN * 100

#ifndef RUDDER_MIX
 # define RUDDER_MIX           0.5f
#endif


//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////
// DEBUGGING
//////////////////////////////////////////////////////////////////////////////
//////////////////////////////////////////////////////////////////////////////

//////////////////////////////////////////////////////////////////////////////
// Dataflash logging control
//

#ifndef LOGGING_ENABLED
 # define LOGGING_ENABLED                ENABLED
#endif

#define DEFAULT_LOG_BITMASK   0xffff


//////////////////////////////////////////////////////////////////////////////
// Navigation defaults
//
#ifndef WP_RADIUS_DEFAULT
 # define WP_RADIUS_DEFAULT              90
#endif

#ifndef LOITER_RADIUS_DEFAULT
 # define LOITER_RADIUS_DEFAULT 60
#endif

#ifndef ALT_HOLD_HOME
 # define ALT_HOLD_HOME 100
#endif
#define ALT_HOLD_HOME_CM ALT_HOLD_HOME*100

#ifndef USE_CURRENT_ALT
 # define USE_CURRENT_ALT FALSE
#endif

#ifndef INVERTED_FLIGHT_PWM
 # define INVERTED_FLIGHT_PWM 1750
#endif

#ifndef PX4IO_OVERRIDE_PWM
 # define PX4IO_OVERRIDE_PWM 1750
#endif

//////////////////////////////////////////////////////////////////////////////
// Developer Items
//

#ifndef SCALING_SPEED
 # define SCALING_SPEED          15.0
#endif

// use this to completely disable the CLI. We now default the CLI to
// off on smaller boards.
#ifndef CLI_ENABLED
#define CLI_ENABLED ENABLED
#endif

// use this to disable geo-fencing
#ifndef GEOFENCE_ENABLED
 # define GEOFENCE_ENABLED ENABLED
#endif

// pwm value on FENCE_CHANNEL to use to enable fenced mode
#ifndef FENCE_ENABLE_PWM
 # define FENCE_ENABLE_PWM 1750
#endif

// a digital pin to set high when the geo-fence triggers. Defaults
// to -1, which means don't activate a pin
#ifndef FENCE_TRIGGERED_PIN
 # define FENCE_TRIGGERED_PIN -1
#endif

// if RESET_SWITCH_CH is not zero, then this is the PWM value on
// that channel where we reset the control mode to the current switch
// position (to for example return to switched mode after failsafe or
// fence breach)
#ifndef RESET_SWITCH_CHAN_PWM
 # define RESET_SWITCH_CHAN_PWM 1750
#endif

// OBC Failsafe enable
#ifndef OBC_FAILSAFE
#define OBC_FAILSAFE ENABLED
#endif

#define HIL_SUPPORT ENABLED

//////////////////////////////////////////////////////////////////////////////
// Parachute release
#ifndef PARACHUTE
#define PARACHUTE ENABLED
#endif

/*
  build a firmware version string.
  GIT_VERSION comes from Makefile builds
*/
#ifndef GIT_VERSION
#define FIRMWARE_STRING THISFIRMWARE
#else
#define FIRMWARE_STRING THISFIRMWARE " (" GIT_VERSION ")"
#endif
